### PR TITLE
Add v1.0.0 Conformance Report of Envoy Gateway

### PR DESCRIPTION
**Pull request by Xunzhuo**
**What type of PR is this?**

/area conformance

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes-sigs/gateway-api/issues/2557

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by arkodg — APPROVED** [review]
/approve
thanks !

**Review by tao12345666333 — APPROVED** [review]

**Comment by robscott** [comment]
I'm probably missing something obvious, but how is it possible to pass all tests with v1.0.0 per https://github.com/kubernetes-sigs/gateway-api/pull/2535/files#r1378650459?

**Comment by arkodg** [comment]
@robscott the current code in v1.0.0 made an extra assertion that the Route is not `Accepted` . This is true for Envoy Gateway.
However there is nothing in the spec to enforce that, so that check was removed. ptal at https://github.com/kubernetes-sigs/gateway-api/pull/2548 for more context
tldr - Envoy Gateway will pass all tests with or without that assertion

**Comment by robscott** [comment]
Thanks @arkodg! I guess the question was how that test could pass (Accepted == false) at the same time as a nearly identical test that requires Accepted to be true: https://github.com/kubernetes-sigs/gateway-api/blob/341c46e29f16501f301abf04313b8d2cf6e48e0f/conformance/tests/httproute-invalid-backendref-nonexistent.go#L48

Just trying to understand the difference here when this seems to be a blocker for many implementations to pass conformance on v1.0.0. Maybe Envoy Gateway is making a distinction between the two test cases that other implementations are not?

**Comment by arkodg** [comment]
Envoy Gateway defines its own custom reason (for the Route) for this case/test- `NoReadyListeners`

      - lastTransitionTime: "2023-10-27T01:38:14Z"
        message: There are no ready listeners for this parent ref
        observedGeneration: 1
        reason: NoReadyListeners
        status: "False"
        type: Accepted
      - lastTransitionTime: "2023-10-27T01:38:14Z"

The listener is not ready in this test because the cert is invalid

      - lastTransitionTime: "2023-10-27T01:38:14Z"
        message: Secret gateway-conformance-infra/does-not-exist does not exist.
        observedGeneration: 2
        reason: InvalidCertificateRef
        status: "False"
        type: ResolvedRefs
      - lastTransitionTime: "2023-10-27T01:38:14Z"
        message: Listener is invalid, see other Conditions for details.
        observedGeneration: 2
        reason: Invalid
        status: "False"
        type: Programmed

**Comment by robscott** [comment]
Got it, makes sense, thanks for the explanation @arkodg!

/hold cancel
/approve
